### PR TITLE
Add rubygem-racc

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by ekohl — APPROVED** [review]

**Comment by ehelms** [comment]
This now includes some additional exclude and updates the gem2rpm template

File: packages/foreman/rubygem-racc/rubygem-racc.spec [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+# start specfile generated dependencies
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}ruby
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby-devel
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: tree
```
Comment (ekohl) [inline packages/foreman/rubygem-racc/rubygem-racc.spec:24]: How does it generate this dependency? Feels odd.

File: packages/foreman/rubygem-racc/rubygem-racc.spec [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+%dir %{gem_instdir}
+%{_bindir}/racc
+%exclude %{gem_instdir}/ext
+%{gem_extdir_mri}
+%license %{gem_instdir}/COPYING
+%exclude %{gem_instdir}/TODO
+%{gem_instdir}/bin
+%{gem_libdir}
+%exclude %{gem_instdir}/sample
+%exclude %{gem_instdir}/web
```
Comment (ekohl) [inline packages/foreman/rubygem-racc/rubygem-racc.spec:108]: In https://src.fedoraproject.org/rpms/rubygem-racc/blob/rawhide/f/rubygem-racc.spec they ship it in the doc subpackage

File: gem2rpm/scl.spec.erb [hunk 3]
```diff
@@ -12,6 +12,9 @@ config.rules[:ignore] << '.document'
 config.rules[:ignore] << 'Appraisals'
 config.rules[:ignore] << 'gemfiles'
 config.rules[:ignore] << 'appveyor.yml'
+config.rules[:ignore] << 'TODO'
+config.rules[:ignore] << 'sample'
+config.rules[:ignore] << 'web'
```
Comment (ekohl) [inline gem2rpm/scl.spec.erb:17]: In https://src.fedoraproject.org/rpms/rubygem-racc/blob/rawhide/f/rubygem-racc.spec they're shipped in the doc pacakge. I'm not sure we should ignore them.

**Review by ekohl — COMMENTED** [review]
https://src.fedoraproject.org/rpms/rubygem-racc/blob/rawhide/f/rubygem-racc.spec also has a build requires on gcc. I wonder why it's not needed here.

File: packages/foreman/rubygem-racc/rubygem-racc.spec (hunk 1 above)
Comment (ehelms) [reply packages/foreman/rubygem-racc/rubygem-racc.spec:24]: Oops, that was from my debugging.

File: gem2rpm/scl.spec.erb [hunk 4]
```diff
@@ -6,12 +6,15 @@ SCL_PREFIXES = JSON.load(File.read(File.join(root, 'scl_prefixes.json')))
 LICENSES = JSON.load(File.read(File.join(root, 'licenses.json')))
 
 config.rules[:doc] << /\/?CHANGES/
+config.rules[:doc] << 'web'
```
Comment (ekohl) [inline gem2rpm/scl.spec.erb:9]: I still feel like web (and sample below) are too generic and have a risk of causing problems in the future. @evgeni?

Comment (evgeni) [reply gem2rpm/scl.spec.erb:9]: I am not versed enough in the standard (or well, in this case less standard) layouts of gems and what can be hidden in these folders.

That said, the content of the `web` folder doesn't look like docs directly -- these are templates that need processing and look kinda useless w/o it. So for `racc`, I lean towards `%exclude`ing it completely.

And then I know no examples of `web` being anything, so maybe it should not be part of the template either?

`sample` on the other hand actually looks like useful examples, which I'd consider documentation? But the `sample` name is indeed a tad off, I'd usually put those in `examples`.

**Comment by evgeni** [comment]
> also has a build requires on gcc. I wonder why it's not needed here.

probably because we add it to the default build group, while Fedora stopped doing that a few releases back?

https://github.com/theforeman/tool_belt/blob/548c363fe61a68438cb3db3b6d0a9673f2ae7e48/configs/foreman/2.5.yaml#L49

File: packages/foreman/rubygem-racc/rubygem-racc.spec [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%doc %{gem_instdir}/ChangeLog
+%doc %{gem_instdir}/README.ja.rdoc
+%doc %{gem_instdir}/README.rdoc
+%{gem_instdir}/Rakefile
+%doc %{gem_instdir}/rdoc
+%{gem_instdir}/web
```
Comment (evgeni) [inline packages/foreman/rubygem-racc/rubygem-racc.spec:117]: As mentioned in the other comment, I think this is not useful documentation and we should not ship it.

(those are the sources that generate https://i.loveruby.net/en/projects/racc/, which hasn't been updated for the latest releases either)

File: packages/foreman/rubygem-racc/rubygem-racc.spec [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+%files
+%dir %{gem_instdir}
+%{_bindir}/racc
+%exclude %{gem_instdir}/ext
+%{gem_extdir_mri}
+%license %{gem_instdir}/COPYING
+%exclude %{gem_instdir}/TODO
+%{gem_instdir}/bin
+%{gem_libdir}
+%exclude %{gem_instdir}/sample
```
Comment (evgeni) [inline packages/foreman/rubygem-racc/rubygem-racc.spec:106]: IMHO examples belong to the doc package

**Comment by ehelms** [comment]
Updated based on last round of comments

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
@ehelms seems you forgot something ;)

https://ci.theforeman.org/job/foreman-packaging-rpm-nightly-release/931/console

**Pull request merged**